packages/client/src/sigstore-utils.ts

```diff
@@ -30,45 +39,50 @@ export async function createDSSEEnvelope(
     signer: SignerFunc;
   }
 ): Promise<SerializedEnvelope> {
-  // Pre-authentication encoding to be signed
-  const paeBuffer = dsse.preAuthEncoding(payloadType, payload);
-
-  // Get signature and verification material for pae
-  const sigMaterial = await options.signer(paeBuffer);
-
-  const envelope: sigstore.Envelope = {
-    payloadType,
-    payload,
-    signatures: [
-      {
-        keyid: sigMaterial.key?.id || '',
-        sig: sigMaterial.signature,
-      },
-    ],
-  };
-
-  return sigstore.Envelope.toJSON(envelope) as SerializedEnvelope;
+  const bundler = createBundleBuilder('dsseEnvelope', {
+    signer: options.signer,
+    tlogUpload: false,
+  });
+  const bundle = await bundler.create({ data: payload, type: payloadType });
+  return envelopeToJSON(bundle.content.dsseEnvelope);
```

Comment:
Note how the `createDSSEEnvelope` and `createRekorEntry` functions we export here (exclusively for use by the npm registry) are now implemented by calls to the `DSSEBundleBuilder` and the `RekorWitness` (both exported from `@sigstore/sign`). 

In the future, we can probably just move these functions into the places in the registry where they are needed.

Reply:
Nice! Def up for migrating this to the registry and adding these utils the deprecations for 2.0 🙌